### PR TITLE
download tab - label NS (not sequenced) and WT (wild type) for mutations, and NP (not profiled) for cna, mrna, prot when necessary

### PR DESCRIPTION
**Pull request by adamabeshouse**
Fixes https://github.com/cBioPortal/cbioportal/issues/3972

[Example link](https://www.cbioportal.org/results/download?Action=Submit&RPPA_SCORE_THRESHOLD=2.0&Z_SCORE_THRESHOLD=2.0&cancer_study_id=coadread_tcga_pub&cancer_study_list=msk_impact_2017&case_ids=msk_impact_2017%3AP-0000004-T01-IM3%2Bmsk_impact_2017%3AP-0000015-T01-IM3%2Bmsk_impact_2017%3AP-0000023-T01-IM3%2Bmsk_impact_2017%3AP-0000024-T01-IM3&case_set_id=-1&data_priority=0&gene_list=OCA2%2520PTPN22%2520TP53&gene_set_choice=user-defined-list&geneset_list=%20&genetic_profile_ids_PROFILE_COPY_NUMBER_ALTERATION=msk_impact_2017_cna&genetic_profile_ids_PROFILE_MUTATION_EXTENDED=msk_impact_2017_mutations&profileFilter=0&tab_index=tab_visualize)

### CODE REVIEW

File: src/pages/resultsView/download/DownloadUtils.ts [hunk 1]
```diff
@@ -234,14 +237,23 @@ export function generateMutationData(
 export function generateMutationDownloadData(
     sampleAlterationDataByGene: { [key: string]: ExtendedAlteration[] },
     samples: Sample[] = [],
-    genes: Gene[] = []
+    genes: Gene[] = [],
+    isSampleProfiled: (
```
Comment (alisman) [inline src/pages/resultsView/download/DownloadUtils.ts:241]: isSampleProfiledFunction?

File: src/pages/resultsView/download/DownloadUtils.ts [hunk 2]
```diff
@@ -500,12 +518,45 @@ export function generateGenericAssayRowsByUniqueSampleKey(
...
+            return false;
+        }
+    };
+}
+
 export function generateDownloadData(
     sampleAlterationDataByGene: { [key: string]: ExtendedAlteration[] },
     samples: Sample[] = [],
     genes: Gene[] = [],
+    isSampleProfiled: (
```
Comment (alisman) [inline src/pages/resultsView/download/DownloadUtils.ts:551]: isSampleProfiledFunction?

File: src/pages/resultsView/download/DownloadUtils.ts [hunk 3]
```diff
@@ -536,10 +587,21 @@ export function generateDownloadData(
...
+                    rowData.uniqueSampleKey,
+                    rowData.studyId,
+                    gene
+                )
+            ) {
+                formattedValue = notProfiledString;
+            } else {
+                formattedValue = formatData
+                    ? formatData(rowData.alterationData[gene]) // if provided format with the custom data formatter
+                    : rowData.alterationData[gene].join(' ') ||
```
Comment (alisman) [inline src/pages/resultsView/download/DownloadUtils.ts:602]: not sure i understand this or statement.    ```rowData.alterationData[gene].join(' ')```
is somehow falsy?

Comment (alisman) [reply src/pages/resultsView/download/DownloadUtils.ts:602]: i think would be better if you just accomplished with ifs.

**Review by alisman — APPROVED** [review]

File: src/pages/resultsView/download/DownloadUtils.ts (hunk 3 above)
Comment (adamabeshouse) [reply src/pages/resultsView/download/DownloadUtils.ts:602]: good catch, this snippet is a nightmare

**Review by jjgao — APPROVED** [review]

**Pull request merged**